### DISCUSSION TITLE
Store path for images pasted

### DISCUSSION THREAD

**Discussion by beccon4** [freeplane/freeplane#2076] [Ideas]
I constantly copy Mindmaps to make sure not to lose data and to have a history. As long as they do not contain images there is no problem at all if the file name changes but if they do the next image pasted gets into a new folder. 

Would it be possible to either:
- set a fixed directory  specific to the mindmap where all new images/files pasted go to. (this would also be of use to have a single folder for various mindmaps)
- tidy up an existing configuration by rearranging images into a new folder

**Comment by Salamimonster** [comment]
I just realized, how problematic images in a freeplane mindmap are.

If one inserts an image into a mindmap, it creates a folder with the name <mindmapname>_files in the same folder the mm-file lies and stores the image in there.
So, if you want to move the mindmap - you have to move the folder with it. Uncool. But it gets worse. 
If you save the Mindmap using a different name (for example to create backup versions) and you insert an other image into that renamed mindmap, there is a new "<mindmapname>_files"-folder created. And if you want to move this mindmap - you have to move both <mindmapname>_files-folders... 

Is there any other way, to insert images into a mindmap, without having the trouble of creating extra folders for the images?

**Reply by macmarrum** [reply]
There are several ways to add an image to a mind map in Freeplane. I assume you do a drag-and-drop.

If your images are already in your hard drive, you can copy them in you file manager (`Ctrl+C`) and paste in Freeplane (`Ctrl+V`), as opposed to drag-and-drop. You will get a link to the original image file.

The link can be absolute or relative, controlled by `Preferences…->Environment->Hyperlink types->Links`.

You could put your image files in the same directory as your mind map, set `Links` to `Relative`, then copy-paste image files from your file manager to Freeplane. This way you can rename your mind map and keep the links unchanged. And when you move your mind map, you can move (or copy) your image files with it.

---

If you have your image files linked to nodes, you can try [copyLinkedFilesToClipboard.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/refs/heads/main/src/scripts/copyLinkedFilesToClipboard.groovy) to copy them from selected nodes as if they were selected in your file manager and copied there.

- https://github.com/freeplane/freeplane/discussions/1467

**Comment by Salamimonster** [comment]
Thank you for explaining @macmarrum 
Is there a way to embed the images inside the .mm file? So that there is only one file to move around?

**Reply by euu2021** [reply]
see: https://github.com/freeplane/freeplane/discussions/1106

and 

https://github.com/freeplane/freeplane/issues/1089
